agent/serviceconnect/load.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package serviceconnect
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
+	log "github.com/cihub/seelog"
+	"github.com/docker/docker/api/types"
+	"github.com/pkg/errors"
+)
+
+var (
+	defaultAgentContainerImageName = "appnet_agent"
+	defaultAgentContainerTag       = "service_connect.v1"
+)
+
+// Loader defines an interface for loading the appnetAgent container image. This is mostly
+// to facilitate mocking and testing of the LoadImage method
+type Loader interface {
+	LoadImage(ctx context.Context, dockerClient dockerapi.DockerClient) (*types.ImageInspect, error)
+	IsLoaded(dockerClient dockerapi.DockerClient) (bool, error)
+}
+
+type loader struct {
+	AgentContainerImageName   string
+	AgentContainerTag         string
+	AgentContainerTarballPath string
+}
+
+// New creates a new pause image loader
+func New() Loader {
+	return &loader{
+		AgentContainerImageName:   defaultAgentContainerImageName,
+		AgentContainerTag:         defaultAgentContainerTag,
+		AgentContainerTarballPath: defaultAgentContainerTarballPath,
+	}
+}
+
+// This function uses the DockerClient to inspect the image with the given name and tag.
+func getAgentContainerImage(name string, tag string, dockerClient dockerapi.DockerClient) (*types.ImageInspect, error) {
+	imageName := fmt.Sprintf("%s:%s", name, tag)
+	log.Debugf("Inspecting appnet agent container image: %s", imageName)
```

Comment:
nit. Could we use `logger.Debugf()` as we are gradually migrating to logger. Thanks.